client/src/main/java/io/cloudsoft/winrm4j/client/WinRmClient.java

```diff
@@ -154,10 +170,19 @@ public int command(String cmd, Writer out, Writer err) {
         }
     }
 
+    // TODO fix CXF to not set a wrong action https://issues.apache.org/jira/browse/CXF-4647
+    private void setActionToContext(BindingProvider bp, String action) {
+        AttributedURIType attrUri = new AttributedURIType();
+        attrUri.setValue(action);
+        ((AddressingProperties)bp.getRequestContext().get("javax.xml.ws.addressing.context")).setAction(attrUri);
```

Comment:
Caching the action header is probably due to https://github.com/cloudsoft/winrm4j/pull/12/files#diff-f1257a98db8070323be7432bc21c48daR270. It's re-using the same object, but initializing the action only the first time.